### DISCUSSION TITLE
Support link to section/heading

### DISCUSSION THREAD

**Discussion by TheColorman** [alangrainger/share-note#54] [Vote for upcoming features]
In obsidian, you can link to a section of a different note using `[[Note#Heading]]`, and clicking on the link will take you to the note and scroll down to the heading.
I realise that the built-in browser heading functionality is currently being used for the decryption key, which isn't really something that can be changed currently, so I guess the best solution to this is either adding a deliminator in the hashcode to separate key and heading, or make use of query parameters, though both of these solutions would probably require client-side javascript for the scrolling/highlighting behavior.

Edit: clarify use for cross-note links

**Comment by TheColorman** [comment]
As for a use case for this feature, I have notes where I refer to the Gaussian distribution by linking to the "Normal Distribution" heading in my "list of distributions" note. This note does not contain any reference to the name Gaussian (a problem on my part), so it is not entirely intuitive to look for the 'Normal Distribution' section of the note, instead of 'Gaussian Distribution'.

**Comment by AbdelrhmanUZaki** [comment]
# It is working with headers

https://github.com/alangrainger/share-note/assets/99971020/c9d0cdf3-5b31-47c2-a274-7b54eed147af

**Reply by TheColorman** [reply]
It works within the same file, but if you link to a heading in a different note it won't scroll down to it.

**Comment by AbdelrhmanUZaki** [comment]
> It works within the same file, but if you link to a heading in a different note it won't scroll down to it.
I think this is normal, there is no relation between them

try to do something like
```
for more info go to [[https://note_1_url#section 1| Section 1]] in home page
```

**Reply by TheColorman** [reply]
The problem is, when you have encryption enabled, the stuff after the hash (#) in the URL is the decryption key. Normally, this represents the heading you would scroll to, but since it's already used for the decryption key, the URLs to other notes don't include the stuff after the hash

**Reply by AbdelrhmanUZaki** [reply]
honestly, I don't know

**Comment by janinko** [comment]
I have a set of heavily interlinked notes and the scrolling to a heading works only sometimes on the same page and never on different one.

**Reply by janinko** [reply]
When I have document Foo and in that document ues `[[#Header1|Bar]]` it will create:
```
<a data-href="#Header1" class="internal-link" rel="noopener" onclick="document.querySelectorAll('[data-heading=&quot;Header1&quot;]')[0].scrollIntoView(true)">Bar</a>
```
and the jumping to header works.

However if I have link ``[[Foo#Header1|Bar]]`, then it will create:
```
<a data-href="Foo#Header1" href="https://share.note.sx/aaa#bbb" class="internal-link" rel="noopener">Bar</a>
```
and the jumping to header doesn't work. Note that it's still within the same document.

**Reply by janinko** [reply]
Also for the different note linking, could there be some additional data with the header name be passed in the part after "#" that has the key, that could be used on the other page to then `scrollIntoView`?
